### PR TITLE
Cover: Add repeated background option

### PR DESCRIPTION
**Pull request by retrofox**
## Description

This PR adds the `Repeated Background` option to the `core/cover` block.

ezgif com-gif-maker (1)

## How has this been tested?

* Create a cover block
* Add an image as the background
* Open the Block settings sidebar
* Confirm you see the `Repeated Background` just below to `Fixed Background`

* Play with different combinations

## Screenshots 

image

image

## [Video Demos](https://cloudup.com/i9ouIWrRslS)

| video 01 | video 02 |
----- |----
[](https://cloudup.com/iNfYd36IURA) | [](https://cloudup.com/i9ouIWrRslS)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/cover/edit.js [hunk 1]
```diff
@@ -309,7 +316,9 @@ function CoverEdit( {
 
 	const hasBackground = !! ( url || overlayColor.color || gradientValue );
 	const showFocalPointPicker =
-		isVideoBackground || ( isImageBackground && ! hasParallax );
+		isRepeated ||
```
Comment (ntsekouras) [inline packages/block-library/src/cover/edit.js:319]: `isRepeated` makes sense for image only, no? Also it 'cancels' the previous logic about focal point below.

So depending on whether we want to show the `FocalPointPicker` on `repeated` images the below condition (`( isImageBackground && ! hasParallax )` should change to reflect the wanted behavior.

**Comment by paaljoachim** [comment]
This is a very cool PR! Thank you for creating the various videos! They show a nice overview of the creativity that one can add by repeating a pattern. I have also tested the PR.

One thing that comes to mind is how the Customizer handles the Background Image.
It uses a Preset drop down.

Screen Shot 2020-10-12 at 13 40 43

Here is an example of adding the same drop down to Gutenberg Cover block.
Cover-block-preset-options-dropdown

I will ping:
@mapk @ItsJonQ 
As we need some more input on the UI.

File: packages/block-library/src/cover/edit.js (hunk 1 above)
Comment (retrofox) [reply packages/block-library/src/cover/edit.js]: > isRepeated makes sense for image only, no?

correct.

> Also it 'cancels' the previous logic about focal point below.

I'd say _updates_ ;-)

> ...to reflect the wanted behavior.

The wanted behavior is being able to set a repeated background, may set it as fixed, and why not change the focal point of it. That works for me.

**Review by mapk — APPROVED** [review]
I just tested this PR and it works well. 👍 

repeat

I do like @paaljoachim's comment about including a few more options as the Customizer does, but I don't want to hold up this PR. Maybe we can get this in and then work toward those other options in another PR?

**Comment by retrofox** [comment]
> I do like @paaljoachim's comment about including a few more options as the Customizer does...

Yes. I do like it too, but wondering whether it should be adapted to the final user. I mean maybe it isn't totally clear what means every background option.

> ..., but I don't want to hold up this PR.

👍 

> Maybe we can get this in and then work toward those other options in another PR?

Sounds good to me. Thanks.

**Comment by noahtallen** [comment]
@retrofox I think this may have broken unit tests for the repo -- I think `isRepeated` needs to be added to the block fixtures for the unit tests

**Comment by noahtallen** [comment]
Fix in #26044, I think

**Comment by retrofox** [comment]
> @retrofox I think this may have broken unit tests for the repo -- I think `isRepeated` needs to be added to the block fixtures for the unit tests

Sorry, my bad. I didn't realize it. 🤦 Thank you very much for fixing it. It won't happen again.

**Comment by marekhrabe** [comment]
Awesome feature btw! :)

**Pull request merged**